blog-samples/CSharp/FacebookHandover/Secondary/Bots/SecondaryBot.cs

```diff
@@ -0,0 +1,143 @@
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var text = turnContext.Activity.Text;
+
+            switch (text)
+            {
+                case OPTION_PASS_PRIMARY_BOT:
+                    await turnContext.SendActivityAsync("Passing thread control to the primary app...");
+                    // A null target app ID will automatically pass control to the primary receiver
+                    await FacebookThreadControlHelper.PassThreadControlAsync(_configuration["FacebookPageToken"], null, turnContext.Activity.From.Id, text);
+                    break;
+
+                case OPTION_TAKE_THREAD_CONTROL:
+                    // Do nothing because the primary receiver should react to this instead
+                    break;
+
+                default:
+                    await ShowChoices(turnContext, cancellationToken);
+                    break;
+            }
+        }
+
+        protected override async Task OnConversationUpdateActivityAsync(ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("SecondaryBot - Processing a ConversationUpdate Activity.");
+
+            var facebookPayload = (turnContext.Activity.ChannelData as JObject)?.ToObject<FacebookPayload>();
+
+            if (facebookPayload != null)
+            {
+                if (facebookPayload.PassThreadControl != null)
+                {
+                    await turnContext.SendActivityAsync($"Thread control is now passed to {facebookPayload.PassThreadControl.NewOwnerAppId} with the message: {facebookPayload.PassThreadControl.Metadata}");
+                    await ShowChoices(turnContext, cancellationToken);
+                }
+                else if (facebookPayload.TakeThreadControl != null)
+                {
+                    await turnContext.SendActivityAsync($"Thread control is now passed to the primary app with the message: {facebookPayload.TakeThreadControl.Metadata}."
+                        + $" Previous thread owner: {facebookPayload.TakeThreadControl.PreviousOwnerAppId}."
+                        + $" Send a message to have the primary app respond.");
+                }
+            }
+
+            await base.OnConversationUpdateActivityAsync(turnContext, cancellationToken);
+        }
+        protected override async Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("SecondaryBot - Processing an Event Activity.");
+
+            // Analyze Facebook payload from EventActivity.Value
+            await ProcessFacebookMessage(turnContext, turnContext.Activity.Value, cancellationToken);
+        }
+
+        private async Task<bool> ProcessFacebookMessage(ITurnContext turnContext, object data, CancellationToken cancellationToken)
```

Comment:
This method is defined as private, and only called by OnEventActivityAsync... it would make sense to just move the call to ProcessStandbyPayload into OnEventActivityAsync.